envoy/api/v2/core/protocol.proto

```diff
@@ -33,6 +33,20 @@ message Http1ProtocolOptions {
   // envoy as their HTTP proxy. In Unix, for example, this is typically done by setting the
   // *http_proxy* environment variable.
   google.protobuf.BoolValue allow_absolute_url = 1;
+
+  // [#not-implemented-hide:]
+  // Handle incoming HTTP/1.0 requests.
+  // This is off by default, and not fully HTTP/1.0 compliant. There is planned
+  // support for HTTP/1.0 style connect logic, dechunking, and handling lack of
+  // client host iff the default_host field is configured.
+  // Any requests for further 1.0 compatibility can be added here:
+  // https://github.com/envoyproxy/envoy/issues/170
+  google.protobuf.BoolValue http_1_0_compatibility_mode = 2;
+
+  // [#not-implemented-hide:]
+  // A default host. This is highly suggested if http_1_0_compatibility_mode is
+  // turned on as Envoy does not fully support HTTP/1.0 without a Host header.
+  string default_host = 3;
```

Comment:
Why is this needed? Can we just ignore this for now?

Reply:
The request I got internally was "we need to handle http/1.0 without a host header" so it's either this or make our code base resistant to an empty host.  I think this is preferable.

Reply:
Hah. OK. SGTM. :)